### PR TITLE
📖 Update testing docs

### PR DESCRIPTION
**Pull request by alexander-demichev**
**What this PR does / why we need it**:

Remove deprecated `test/helpers` instructions and replace with new steps using `internal/envtest`.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/book/src/developer/testing.md [hunk 1]
```diff
@@ -204,11 +207,13 @@ func TestMain(m *testing.M) {
 	// Bootstrapping test environment
```
Comment (sbueringer) [inline docs/book/src/developer/testing.md:207]: The example seems redundant to me.

**Comment by stmcginnis** [comment]
Looks good once @sbueringer's comments are addressed.

**Comment by stmcginnis** [comment]
/lgtm

Thanks!

**Review by fabriziopandini — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**